app/react/translations/edit/editors/HTMLEditor/RichEditor.js

```diff
@@ -0,0 +1,117 @@
+/* @flow */
+import React from 'react';
+import { Editor, EditorState, RichUtils, ContentState, convertFromHTML } from 'draft-js';
+import type { EditorState as EditorStateType } from 'draft-js';
+import { stateToHTML } from 'draft-js-export-html';
+
+import BlockStyleControls from './components/BlockStyleControls';
+import InlineStyleControls from './components/InlineStyleControls';
+import ToggleHTMLButton from './components/ToggleHTMLButton';
+
+type PropTypes = {
+  onChange: Function,
+  toggleRawEdit: Function,
+  value: string,
+  editAsRaw: boolean,
+  fieldInfo: Object
+}
+
+type StateTypes = {
+  editorState: EditorStateType
+}
+
+const getBlockStyle = block =>
+  (block.getType() === 'blockquote') && 'RichEditor-blockquote';
+
+export default class RichEditor extends React.Component {
+  constructor(props: PropTypes) {
+    super(props);
+    const markup = convertFromHTML(props.value);
+    const editorState: EditorStateType = ContentState.createFromBlockArray(
+      markup.contentBlocks,
+      markup.entityMap
+    );
+    this.state = { editorState: EditorState.createWithContent(editorState) };
+    this.focus = () => this.editor.focus();
+  }
+
+  // eslint-disable-next-line react/sort-comp
+  editor: HTMLElement
+  focus: Function
+  state: StateTypes
+  props: PropTypes
+  onTab: Function
+  onChange: Function
+
+  handleKeyCommand = (command: string) => {
+    const { editorState } = this.state;
+    const newState = RichUtils.handleKeyCommand(editorState, command);
+    if (newState) {
+      this.handleChange(newState);
+      return true;
+    }
+    return false;
+  }
+  onTab = (e: Event) => {
```

Comment:
I'll not mention return type of function everywhere. Flow can interpolate correct return, but it's nice documenting feature, from my perspective.
```javascript
(e: Event): void => {